bioimageio/core/prediction.py

```diff
@@ -211,75 +229,85 @@ def load_tile(tile):
     for outer_tile, inner_tile, local_tile in tiles:
         inp, pad_right = load_tile(outer_tile)
         out = predict_with_padding(prediction_pipeline, inp, padding, pad_right)
+        assert len(out) == 1
+        out = out[0]
         output[inner_tile] = out[local_tile]
 
 
 #
 # prediction functions
-# TODO support models with multiple in/outputs
 #
 
 
-def predict(prediction_pipeline, inputs):
-    if isinstance(inputs, np.ndarray):
+def predict(prediction_pipeline, inputs) -> List[xr.DataArray]:
+    if not isinstance(inputs, (tuple, list)):
         inputs = [inputs]
-    if len(inputs) > 1:
-        raise NotImplementedError(len(inputs))
-    axes = tuple(prediction_pipeline.input_axes)
-    tagged_data = [xr.DataArray(ipt, dims=axes) for ipt in inputs]
+
+    tagged_data = [xr.DataArray(ipt, dims=axes) for ipt, axes in zip(inputs, prediction_pipeline.input_axes)]
     return prediction_pipeline.forward(*tagged_data)
 
 
-def predict_with_padding(prediction_pipeline, inputs, padding, pad_right=True):
-    if isinstance(inputs, (np.ndarray, xr.DataArray)):
+def predict_with_padding(prediction_pipeline, inputs, padding, pad_right=True) -> List[xr.DataArray]:
```

Comment:
I think we should also enable passing padding as a list here; if different input tensors are given they might also need to be padded differently. 
So I would add something like `if not isinstance(padding, (tuple, list)): padding = [padding]` and then also iterate over padding in line 255.